python/tk_desktop/desktop_engine_site_implementation.py

```diff
@@ -57,28 +57,16 @@ def _init_tabs(self):
         Create desktop tabs from the registered application panels, following
         their specified relative positioning.
         """
-        # make the "Apps" tab appear as a fake panel to fit in the sorting
-        # logic
-        apps_tab = {"callback":   self.desktop_window._register_apps_tab,
-                    "properties": {"position": 0}}
+        # manually register the "Apps" tab for now. Ideally should be an app on
+        # its own.
+        self.desktop_window._register_apps_tab()
 
-        panels = [apps_tab] + self._engine.panels.values()
+        tabs = self._engine.get_setting("tabs", [])
+        commands = self._engine.get_matching_commands(tabs)
 
-        def panel_position(panel):
-            """ Returns the relative position of a panel """
-            # place the panels with unspecified positions at the end
-            LAST_POSITION = float("inf")
-
-            if "properties" not in panel:
-                return LAST_POSITION
-
-            return panel["properties"].get("position", LAST_POSITION)
-
-        ordered_panels = sorted(panels, key=panel_position)
-
-        for panel in ordered_panels:
+        for (_,_,panel_callback) in commands:
             # let the panel show itself through its registered callback
-            panel["callback"]()
+            panel_callback()
```

Comment:
There is a lot of stuff happening here - i think some further explanation in comments would add value.
I would add something along these lines:

```
# panel apps all register commands that when executed will request the engine 
# to show the panel ui. In something like maya or nuke, these commands would appear
# on the shotgun menu and be triggered by the user clicking on them.
#
# here, in desktop, all commands added to the `tabs` configuration setting will
# be executed, a panel is expected to be shown as part of the execution of the 
# panel and when the panel is shown, it is placed in the next available tab slot in
# the desktop, going from left to right.
```